x-pack/auditbeat/module/system/_meta/docs.asciidoc

```diff
@@ -1,22 +1,86 @@
 == System Module
 
-The `system` module ... TODO.
+The `system` module collects various security related information about
+a server. All metricsets send both periodic state information (e.g. all currently
+running processes) and real-time changes (e.g. when a new process starts
+or stops).
 
-The module is implemented for Linux, macOS (Darwin), and Windows.
+The module is fully implemented for Linux, and partially implemented
+for macOS (Darwin).
 
 [float]
 === How it works
 
-TODO
+Each metricset sends two kinds of information: state and events.
+
+State information is sent periodically and (for some metricsets) on startup.
+A state update will consist of one event per object that is currently
+active on the system (e.g. a process). All events belonging to the same state
+update will share the same UUID in `event.id`.
+
+The frequency of state updates can be controlled for all metricsets using the
+`state.period` configuration option. Overrides are available per metricset.
+The default is `12h`.
+
+Event information is sent as the events occur (e.g. a process starts or stops).
+All metricsets are currently using a pull model to retrieve their data.
+The frequency of these pulls is controlled using the `period` configuration
+parameter.
 
 [float]
 === Configuration options
 
-TODO
+This module has some configuration options for controlling its behavior. The
+following example shows all configuration options with their default values for
+Linux.
+
+NOTE: It is recommended to configure some metricsets separately. See below for a
+sample suggested configuration.
 
 [source,yaml]
 ----
 - module: system
+  metricsets:
+    - host
+    - process
+    - socket
+    - user
+  period: 10s
+  state.period: 12h
+  user.detect_password_changes: true
 ----
 
-*`some_option`*:: TODO
+*`period`*:: The frequency at which the metricsets check for changes. For most
+metricsets - esp. `process` and `socket` - a lower value is recommended.
+
+*`state.period`*:: The frequency at which the metricsets send full state information.
+This option can be overridden per metricset using `{metricset}.state.period`.
+
+*`user.detect_password_changes`*:: If the `user` metricset is configured and
+this option is set to `true`, Auditbeat will read password information in `/etc/passwd`
+and `/etc/shadow` to detect password changes. A hash will be kept locally in
+the `beat.db` file to detect changes between Auditbeat restarts.
```

Comment:
Should we add here a warning box to explicitly call out the danger of dictionary attacks on `beat.db`?

Reply:
If someone has root on the system, they have access to the hash directly from passwd, and they have access to a hashed version of the Linux hash in beats.db. Leveraging what's in beats.db is more difficult than leveraging the standard passwd/shadow files. So brute force is not the risk I would warn about.

Perhaps instead we should insist on the fact that this file now contains a copy of the hashes. So users should ensure it's as well scrutinized (e.g. FIM) as /etc/passwd and /etc/shadow, and that the file is not copied around lightly (e.g. for diagnostics, misconfigured backups).

Reply:
How about adding this sentence:

> The `beat.db` file should be readable only by the root user and be treated similar to the shadow file itself.

Reply:
Yes, this works.